### PR TITLE
HDDS-6550. Change log level for exceptions in S3gateway

### PR DESCRIPTION
**Pull request by symious**
## What changes were proposed in this pull request?

In a production cluster, the log level is usually not set to DEBUG, due to the log flushing.

In S3gateway, some exceptions are logged at DEBUG level, which causes the exception not being able to be shown at first place, we need to restart the service and reconfigure the log level, and after the fix the exception, we need to restart the service and set back the log level.

This ticket is to change the log level to ERROR instead of DEBUG to easily find the exception.

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-6550

## How was this patch tested?

uni test.

### CODE REVIEW

File: hadoop-ozone/s3gateway/src/main/java/org/apache/hadoop/ozone/s3/OzoneClientProducer.java [hunk 1]
```diff
@@ -102,12 +103,12 @@ public S3Auth getSignature() {
           signatureInfo.getSignature(),
           awsAccessId);
     } catch (OS3Exception ex) {
-      LOG.debug("Error during Client Creation: ", ex);
+      LOG.error("Error during Client Creation: ", ex);
```
Comment (kerneltime) [inline hadoop-ozone/s3gateway/src/main/java/org/apache/hadoop/ozone/s3/OzoneClientProducer.java:106]: This should not be logged at `error`, a badly configured client can overwhelm the log.

Comment (symious) [reply hadoop-ozone/s3gateway/src/main/java/org/apache/hadoop/ozone/s3/OzoneClientProducer.java:106]: @kerneltime Thanks for the review.

Normally we'd expect the badly configured client to correct their configurations after they received the OS3Exception. But if the client doesn't change and keeps sending requests with bad configuraitons, the server might be in a blind state. With some explicit log, server-side can notice the bad operations and remind the client to correct their operations.

Comment (adoroszlai) [reply hadoop-ozone/s3gateway/src/main/java/org/apache/hadoop/ozone/s3/OzoneClientProducer.java:106]: I agree with @kerneltime: client-side problem should not be logged as error at server-side.  Client receives the error anyway and can act on it.

Comment (symious) [reply hadoop-ozone/s3gateway/src/main/java/org/apache/hadoop/ozone/s3/OzoneClientProducer.java]: Ok, reverted debug log for OS3Exception.

File: hadoop-ozone/s3gateway/src/main/java/org/apache/hadoop/ozone/s3/OzoneClientProducer.java [hunk 2]
```diff
@@ -107,7 +108,7 @@ public S3Auth getSignature() {
     } catch (Exception e) {
       // For any other critical errors during object creation throw Internal
       // error.
-      LOG.debug("Error during Client Creation: ", e);
+      LOG.error("Error during Client Creation: ", e);
       throw wrapOS3Exception(S3ErrorTable.newError(INTERNAL_ERROR, null, e));
```
Comment (adoroszlai) [inline hadoop-ozone/s3gateway/src/main/java/org/apache/hadoop/ozone/s3/OzoneClientProducer.java:112]: I don't think this change is necessary, `S3ErrorTable.newError` also logs internal errors.

https://github.com/apache/ozone/blob/c23aae0d9384a6ad815dc3ad9edeb3cd935c53a5/hadoop-ozone/s3gateway/src/main/java/org/apache/hadoop/ozone/s3/exception/S3ErrorTable.java#L141-L142

Comment (symious) [reply hadoop-ozone/s3gateway/src/main/java/org/apache/hadoop/ozone/s3/OzoneClientProducer.java:112]: Oh, seems it's already being handled. 
Thanks for the remind.

**Comment by symious** [comment]
@kerneltime @adoroszlai 
Since the issue already fixed, I'll close the ticket and PR, thanks for the review.

**Pull request closed without merging**